WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt

```diff
@@ -443,28 +427,25 @@ class OrderDetailViewModel @Inject constructor(
         }
     }
 
-    @VisibleForTesting
-    fun updateOrderStatus(newStatus: String) {
+    private fun updateOrderStatus(newStatus: String) {
         if (networkStatus.isConnected()) {
             launch {
-                if (orderDetailRepository.updateOrderStatus(orderIdSet.id, orderIdSet.remoteOrderId, newStatus)) {
-                    order = order.copy(status = Status.fromValue(newStatus))
-                } else {
-                    onOrderStatusChangeReverted()
-                    triggerEvent(ShowSnackbar(string.order_error_update_general))
+                when (val updateResult = orderDetailRepository.updateOrderStatus(orderIdSet.id, newStatus)) {
+                    is ContinuationWrapper.ContinuationResult.Cancellation -> {
+                        // no-op
+                    }
+                    is ContinuationWrapper.ContinuationResult.Success -> {
+                        if (!updateResult.value) {
+                            triggerEvent(ShowSnackbar(string.order_error_update_general))
+                        }
+                    }
```

Comment:
Hey @wzieba! I have a suggestion for this addition: Wouldn't a nullable cast operation fit best for this scenario instead of using `when` since we have something to do only it's of `Success` type?

Something like this:

```suggestion
                orderDetailRepository.updateOrderStatus(orderIdSet.id, newStatus)
                    .let { it as? ContinuationWrapper.ContinuationResult.Success }
                    ?.takeIf { it.value.not() }
                    ?.let { triggerEvent(ShowSnackbar(string.order_error_update_general)) }
```

WDYT about? This way we could avoid dealing with a `is` case with an empty action

Reply:
Sure thing, for a minute I wasn't very convinced to it but after a while it make sense - I think it's a matter of getting used to `takeIf` for me :)

I've also added tests (3529358) to make sure we do follow previous behavior. Thanks!